plugin/storage/integration/elasticsearch_test.go

```diff
@@ -130,18 +118,33 @@ func (*ESStorageIntegration) initializeESFactory(t *testing.T, allTagsAsFields b
 }
 
 func (s *ESStorageIntegration) initSpanstore(t *testing.T, allTagsAsFields bool) {
-	f := s.initializeESFactory(t, allTagsAsFields)
+	f := s.initializeESFactory(t, []string{
+		fmt.Sprintf("--es.num-shards=%v", 5),
+		fmt.Sprintf("--es.num-replicas=%v", 1),
+		fmt.Sprintf("--es.index-prefix=%v", indexPrefix),
+		fmt.Sprintf("--es.use-ilm=%v", false),
+		fmt.Sprintf("--es.service-cache-ttl=%v", 1*time.Second),
+		fmt.Sprintf("--es.tags-as-fields.all=%v", allTagsAsFields),
+		fmt.Sprintf("--es.bulk.actions=%v", 1),
+		fmt.Sprintf("--es.bulk.flush-interval=%v", time.Nanosecond),
+	}, es.NewFactory)
+	af := s.initializeESFactory(t, []string{
```

Comment:
Hm, one thing that the namespacing logic in the storage factories was doing was focing the archive config inherit the settings from the primary config (not just defaults, but user settings). Which is why in this example mostly only primary settings are provided. Are you able to keep this inheritance?

Reply:
@yurishkuro Where is that happening? I just see the default config being used [here](https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/es/options.go#L113)

Reply:
https://github.com/jaegertracing/jaeger/blob/b04e0bafdfb09eab640c8141c696f0106cf7591a/pkg/cassandra/config/config.go#L124

Reply:
plugin/storage/cassandra/options.go
273:	nsCfg.Configuration.ApplyDefaults(&opt.Primary.Configuration)

Reply:
@yurishkuro I don't see a clean way of doing that with this change since there's no relation between the two configurations anymore. Do you see one?

Reply:
There is still _some_ places in the call stack that knows above both primary and archive storage. For example, `plugin/storage/factory.go`. So we could introduce an API method to tell archive factory to inherit settings from primary factory.

Reply:
@yurishkuro I was thinking the same. But how do we get access to the settings of each individual factory in `plugin/storage/factory.go`?

Reply:
why do we need access? If it's an abstract method of the Factory interface, e.g.
```
type Factory interface {
  ...
  InheritDefaultSettingsFrom(otherFactory Factory)
}
```
then the factory would recognize its own type as `otherFactory` and know how to get to the settings internally.

Reply:
@yurishkuro Are we okay with adding this method to the `Factory` interface? That will be a pretty big change and might affect the `storage_v2` interface as well

Reply:
I was think of only adding it to v1 factories, but it's a good point. Specifically, we need to decide how v2 factories will be initialized in jaeger-v1 in the first place. For jaeger-v2 we don't want to maintain all those flags parsing logic, since v2 only deals with configs (we won't support inheritance of configs there, it can be done via `<<: *commonConfig` in YAML).

To avoid blocking on this now you can introduce another interface with this method and do a runtime check (similar to former ArchiveFactory checks).

Reply:
Especially considering that not all factories even implement this ApplyDefaults, it's only needed for ES/C*